src/compiler/checker.ts

```diff
@@ -4176,11 +4203,15 @@ namespace ts {
          * getExportedTypeFromNamespace('JSX', 'Element') returns the JSX.Element type
          */
         function getExportedTypeFromNamespace(namespace: string, name: string): Type {
-            let namespaceSymbol = getGlobalSymbol(namespace, SymbolFlags.Namespace, /*diagnosticMessage*/ undefined);
-            let typeSymbol = namespaceSymbol && getSymbol(namespaceSymbol.exports, name, SymbolFlags.Type);
+            let typeSymbol = getExportedSymbolFromNamespace(namespace, name);
             return typeSymbol && getDeclaredTypeOfSymbol(typeSymbol);
         }
 
+        function getExportedSymbolFromNamespace(namespace: string, name: string): Symbol {
+            let namespaceSymbol = getGlobalSymbol(namespace, SymbolFlags.Namespace, /*diagnosticMessage*/ undefined);
+            return namespaceSymbol && getSymbol(namespaceSymbol.exports, name, SymbolFlags.Type | SymbolFlags.Value);
+        }
+

```

Comment:
This whole change appears unnecessary. Nothing calls `getExportedSymbolFromNamespace`.